rpc/dcrd.go

```diff
@@ -58,7 +58,8 @@ func (d *DcrdConnect) Close() {
 
 // Client creates a new DcrdRPC client instance. Returns an error if dialing
 // dcrd fails or if dcrd is misconfigured.
-func (d *DcrdConnect) Client(ctx context.Context) (*DcrdRPC, string, error) {
+func (d *DcrdConnect) Client() (*DcrdRPC, string, error) {
+	ctx := context.Background()
```

Comment:
What is the purpose of this context.

If it is to be sure that client can finish what it's doing and cleanly stop in a multi-threaded enviroment, then it needs to take a context and a waitgroup.

If it is to prevent long processes on shutdown from continuing then needs to take a context.

If it is to prevent it from taking to long, it should use `context.WithTimeout`

If it's just to make it work, I think that `context.WithCancel` is recommended, and you do `defer cancel` right after.

In my opinion, the ctx should come from higher up like it was. It *can* be saved in the struct (although rather all the methods just take a separate ctx). All of the rpc calls should take that ctx and further do `context.WithTimeout` with a reasonable timeout, like 5-10 seconds.

Reply:
I updated my golangcli-lint and it does not complain here. So scratch that.

Reply:
I looked a bit further at this and found that `TODO` and `Background` are identical:

https://github.com/golang/go/blob/2a6e13843d5bc0a380ce7081e33db9b636e394f9/src/context/context.go#L199-L218

I'd accept a PR changing from `Background` to `TODO`, f you feel strongly about this, but personally I don't think its worth the churn.

Reply:
I think the reason is having background in the middle of a function is misuse of context.

I could make an argument that this pr is `churn` without the change to TODO at least.

Reply:
Pushed one more change to go from Background to TODO in the rpc clients.

Upon re-reading these comments on a new day, it makes sense. Think I was just stuck in a weird thought pattern on Monday.